src/org/labkey/snd/query/AttributeDataTable.java

```diff
@@ -178,7 +176,23 @@ private void loadPropertyDescriptorCache(User user)
                 packageMap.put(pkgId, packageAttributes);
             }
         }
+        private int getRowCount(DataIteratorBuilder rows, @Nullable Map<Enum,Object> configParameters, BatchValidationException errors)
+        {
+            List<Map<String, Object>> data;
+
+            DataIteratorContext dataIteratorContext = getDataIteratorContext(errors, QueryUpdateService.InsertOption.MERGE, configParameters);
 
+            try
+            {
+                data = _sndService.getMutableData(rows, dataIteratorContext);
+            }
+            catch (IOException e)
+            {
+                return 0;
```

Comment:
If you're keeping this function (see comment below), I would capture any exceptions here to make debugging any IO issues easier in the future.  Should be just this line.

errors.addRowError(new ValidationException(e.getMessage()));